packages/block-library/src/image/edit.js

```diff
@@ -107,8 +111,16 @@ export function ImageEdit( {
 		align,
 		metadata,
 	} = attributes;
+
 	const [ temporaryURL, setTemporaryURL ] = useState( attributes.blob );
 
+	const [ resizeListener, sizes ] = useResizeObserver();
```

Comment:
> Could we rely on that existing CSS class? It's fine that _every placeholder_ gets the is-small changes you've made here, even if only the Image block makes use of them for now. Make sense?

Technically, yes. But in the long run, I'm thinking it could be cleaner and more flexible if this kind of responsive styling was handled solely by the consumer of Placeholder, using [CSS container queries](https://developer.mozilla.org/en-US/docs/Web/CSS/CSS_containment/Container_queries) (rather than a resize observer). That way, consumers wouldn't have to depend on a set of arbitrary breakpoints that Placeholder defines.

I'm not intimately familiar with the requirements of this Image block in particular, but on first glance I would say it could make it simpler and more performant. Could be worth trying here. (This is a non-blocking comment, just suggesting in case it hadn't been considered.)

Reply:
Hi @mirka, this would be a good change, but since it applies more generally to the placeholder component, should we make the change in this PR or address it in a separate issue? Related: https://github.com/WordPress/gutenberg/issues/64288

Reply:
Agreed with Lena — this seems a good fit for CSS Container Queries

Reply:
@vipul0425 I don't have much insight into the current state of things, but it looks like you were able to achieve what you needed in this PR without resize observers or container queries, so yes it seems like something that can be addressed as a separate issue 👍